data-node/src/main/java/org/graylog/datanode/configuration/ConfigurationProvider.java

```diff
@@ -146,16 +164,18 @@ public ConfigurationProvider(Configuration localConfiguration, DataNodeConfig sh
     private void configureInitialAdmin(Path internalUsersFile, String adminUsername, String adminPassword) throws IOException {
         final ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
         final Map<String, Object> map = mapper.readValue(new FileInputStream(internalUsersFile.toFile()), Map.class);
```

Comment:
You can use `TypeReferences.MAP_STRING_OBJECT` instead of `Map.class` to get a generic map return value to avoid the `Map` cast.

Reply:
Thanks! It helps with the top level type, but the access to the admin object will still need casting.

Reply:
Aww, yeah. Well. :smile: